spi/src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/spi/schedule/IntervalSchedule.java

```diff
@@ -72,6 +74,14 @@ public IntervalSchedule(Instant startTime, int interval, ChronoUnit unit) {
         this.clock = Clock.system(ZoneId.systemDefault());
     }
 
+    public IntervalSchedule(StreamInput input) throws IOException {
+        startTime = input.readInstant();
+        interval = input.readInt();
+        unit = input.readEnum(ChronoUnit.class);
+        intervalInMillis = input.readLong();
```

Comment:
Why write/read this instead of constructing based off interval and unit?

Reply:
Great question, this constructor is needed when the input is a stream. This stream is the TCP communication between nodes and helps us serialize and de-serialize information.

Reply:
Had an offline chat with @dbbaughe. Sure makes sense. I'll make the change.